test/rlcIntegration/generated/bodyComplexRest/src/responses.ts

```diff
@@ -35,7 +35,7 @@ export interface BasicGetValid200Response extends HttpResponse {
 
 /** Get complex type {id: 2, name: 'abc', color: 'YELLOW'} */
 export interface BasicGetValiddefaultResponse extends HttpResponse {
-  status: "500";
+  status: "default";
   body: ErrorModelOutput;
```

Comment:
Thanks for preparing this PR!
We all agree that the default `500` might not be a correct error code, but even if we switch to `default` I think it's misleading and hard to leverage the error type indication if we have condition like `response.status == `501` .

I was wondering it it is possible we could have below types:

1. Maybe define all status type: 
2. Define success status type
3. Define non-success type

```typescript
export type AllStatus = '200' | '201' | '202' | '400' | '401' | '404' | '500';
export type GetSuccessStatus = '200';
export type GetDefaultStatus = Exclude<AllStatus, GetSuccessStatus>;
```

Reply:
I was exploring something like this however I found 2 big issues:

1. If we generate a list of all possible status codes, then intellisense suggestions for status codes become useless as the size of the list is huge
![AllStatusCodes](https://user-images.githubusercontent.com/21109913/174149393-0ade4175-c876-45b8-9859-090e86aa5e62.gif)

2. Now narrowing down to all errors also becomes undoable since you will have to write an if check for all the non-success status codes i.e. `if(response.status !== "400" &&  response.status !== "401" &&  response.status !== "402" && response.status !== "403" && response.status !== "404" && response.status !== "405" && /* And so on */)` to get the type narrowed down correctly.

[Here's a sample playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAggNnAysAhsArgZygXigIgEYAGY-AHyOMIqoCZaSBmR4gFlYFZWA2VgdlYAOVgE5GJCTUqFCDGYRYKOC7gr4LBCkQvEy6ZfdKJ15JpSZUm1JjSa0mdJvUSaHXxxWcUXFVxTaKdooOik6KLoRs7lGebN5svmz+bIFswWyhbOFskZwxnJ6c3py+nP6cgZzBnKGc4ZyRPDE8njzePL48-jyBPME8oTzhPJH8Mfye-N78vvz+-IH8wfyh-OH8kUIxQp5C3kK+Qv5CgULBQqFC4UKRojGinqLeor6i-qKBosGioaLhohcBncBmMBjMBgsBisBhsBjsBgcBicBiBkkodFktExEMU2Ki+LhhARhCRhBRhCBBmxmJpELoULoMLocLoCLoSLoKLoQLc2KYYKYEKYUKYMKYcKYCKYSKYKKYQOi2LYYPiyqhyWVcPSyqR2WVQPy2MKxohpWNMMqxoRtWNKMa2OajrB7UdUO6jrh-UdSOGjqB42xkyDENmQZhiyDCNWQZRm2x2wTYP2CahxwTcPOCaR1wTQPu2MehYhr0LMM+hYRv0LKMBtDc7jcxjcZjcFjcVjcNjcdjcDjcTjcLgFjaxlAFrbx44J0+7xPrpIXg8p9ep49p69bDNXnZZq97HNXg559b544Fp9bItPnYlp97MtPg4V9aV45Vr9biVfndSr97mSvoOuT1ka44muBrbmuBnZWuBva2uBg4OuOTqoc2rqoe2Hqod23qof2fqocOgbjsGZGtmGZGdpGZG9jGZGDvG46JixzYpix7bpix3ZZix-a5ixw4FuORaia2paiZ2Faib21aiYOdaUNE7jRMY0RmNEFjRFY0Q2NEdjRA40RONELgqqpY74CqmlTtZM72fp87KYuLmmSuylrtZG7eZp26ebpe6eYZh6eaZJ7KWe1kXpFmnXpFul3pFhmPpFpkvspb7WR+mWad+mW6X+mWGYBmWmSBylgdZEHVZp0HVbpcHVYZiHVaZKHWWhnXqZhnXaThnX6fhnXGURnXmaR1nkVNmlUVNum0VNhkMVNpnMdZrEbepHEbdp3EbfpfEbcZgkbeZInWWJl2aZJl26TJl2GfJl2mUp+D5O4+TGPkZj5BY+RWPkNj5HY+QOPkTj5C4hSfVZhS-XZhSA4QwPOe9rno5DHnvV5xTfaYtDFP9TKE0FOOg6FOOQxF71RaU31CoT8W04DSW06DqW05DGXvVl5TfWqlDlP9mpC0VvOg6VvOQxV71VZU33FIT9WVIDTWVKDrWVJDHXVJ9rSE711T-QN1TA8N1Tg2N1TQ5NtTfdMhNzbUgOLbUoMrbUkPrfUn27ITO31P9+31MDR31ODp31NDF2NN9zyE7djSAw9jSg89jSQ3WADcABQoCQFAADiEDAAAqpgEAAE6YIg6AAMb1xAmC16gGDYHg+DAvgUDkAQoI933XfsPgecF9AJfl5XNcAKJV1XAD2VfIGgWC4FAM8AB713A6AACYQAAPPASBt1gAA0xelxX1e1w3Tctyv7cAHx57nACWAB2wDVwAZigTcr5T1vnXRuzdMAACVm5gAXp-SuUAADeucoBQEwGfTAAAuIBN8a6gIfq3VemBz7IKgAAIwXnvEAWCEEADcUC7wgFgtBVcv4AHMAC+udOEf2-n-ABE9r7T0wAAEQgP-dAcBgBQMwDAuB0AkEoLQYQrBk8cGYDnovZe6DiEoPIZQ6h1dNHUPrhQxhUBP7oAALakOrtnKAljwEoFYWY5hbD2GcO4QfHeKAq7QF-ugT+9dgDv1gVAZxwCa4AAoACUKjBEgPvuA6Rsj4F91UUI0R4jJHJNgZXN+Ji4HACgL4zAEiil4HCWomJb936-ygJEkpZSAB0Sj264BwJ3fA0SgA) with the all status codes idea if you want to play around and see ther UX

In my experience, the most common scenario for error checking is to check if it is an error or not but, no the specific error as the OpenAPI standard doesn't provide a way to define each error, just a general shape.

I'm open to exploring possibilities with you and postponing this for v1.1. However, I'd really like to merge the change to generate "default" instead of "500" for v1.0

Reply:
>In my experience, the most common scenario for error checking is to check if it is an error or not but, no the specific error as the OpenAPI standard doesn't provide a way to define each error, just a general shape.

Yeah, I think the most common scenario is to refer a general shape and we have a swagger extension - [x-ms-error-response](https://github.com/Azure/autorest/tree/main/docs/extensions#x-ms-error-response) which could allow service team to define different error shape.

Reply:
We'd still support x-ms-error-response. Customers will need to check for specific status codes if they want to handle each one specifically.

```ts
if (isUnexpected(response)) {
   if (response.status === "401") {
      throw response.body.authError;
   }

  if (response.status === "400") {
      throw response.body.userError;
  }

   throw response.body.generalError;
} 
```